src/rdkafka_cgrp.c

```diff
@@ -3374,6 +4694,89 @@ static void rd_kafka_cgrp_timeout_scan (rd_kafka_cgrp_t *rkcg, rd_ts_t now) {
 }
 
 
+static void rd_kafka_handle_assign_op (rd_kafka_cgrp_t *rkcg,
+                                       rd_kafka_op_t *rko) {
+        rd_kafka_resp_err_t err = RD_KAFKA_RESP_ERR_NO_ERROR;
+        rd_kafka_error_t *error = NULL;
+
+        if (rkcg->rkcg_flags & RD_KAFKA_CGRP_F_TERMINATE) {
+
+                /* check that the op is valid for the terminating case */
+                if ((rko->rko_u.assign.method ==
+                     RD_KAFKA_ASSIGN_METHOD_ASSIGN ||
+                     rko->rko_u.assign.method ==
+                     RD_KAFKA_ASSIGN_METHOD_INCR_ASSIGN) &&
+                    rko->rko_u.assign.partitions)
+                        err = RD_KAFKA_RESP_ERR__DESTROY;
+                else if (rko->rko_u.assign.method ==
+                         RD_KAFKA_ASSIGN_METHOD_INCR_UNASSIGN &&
+                         rko->rko_u.assign.partitions->cnt !=
+                         rkcg->rkcg_assignment->cnt)
```

Comment:
i don't know. but i'll make this check robust.

Reply:
u.assign.partitions is check for NULL in the first if, but not here.

Reply:
Since u.assign.partitions is set by our code we don't need to check it if we intend it to always be non-NULL, but right now the checks are inconsistent